packages/amazonq/src/extensionNode.ts

```diff
@@ -0,0 +1,92 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { activateAmazonQCommon, amazonQContextPrefix, deactivateCommon } from './extension'
+import { DefaultAmazonQAppInitContext } from 'aws-core-vscode/amazonq'
+import { activate as activateQGumby } from 'aws-core-vscode/amazonqGumby'
+import { ExtContext } from 'aws-core-vscode/shared'
+import { updateDevMode } from 'aws-core-vscode/dev'
+import { CommonAuthViewProvider } from 'aws-core-vscode/login'
+import { isExtensionActive, VSCODE_EXTENSION_ID } from 'aws-core-vscode/utils'
+import { registerSubmitFeedback } from 'aws-core-vscode/feedback'
+import { DevOptions } from 'aws-core-vscode/dev'
+import { Auth } from 'aws-core-vscode/auth'
+import api from './api'
+import { activate as activateCWChat } from './app/chat/activation'
+
+export async function activate(context: vscode.ExtensionContext) {
+    // IMPORTANT: No other code should be added to this function. Place it in one of the following 2 functions where appropriate.
+    await activateAmazonQCommon(context, false)
+    await activateAmazonQNonCommon(context)
+
+    return api
+}
+
+/**
+ * The code in this function is not common, implying it only works in Node.js and not web.
+ * The goal should be for this to not exist and that all code is "common". So if possible make
+ * the code compatible with web and move it to {@link activateAmazonQCommon}.
+ */
+async function activateAmazonQNonCommon(context: vscode.ExtensionContext) {
```

Comment:
Maybe this should be something like:
```suggestion
async function activateAmazonQNode(context: vscode.ExtensionContext) {
```
?